handlers/kannel/handler.go

```diff
@@ -200,10 +201,22 @@ func (h *handler) Send(ctx context.Context, msg courier.MsgOut, clog *courier.Ch
 	}
 
 	var resp *http.Response
+	var respBody []byte
 	if verifySSL {
-		resp, _, err = h.RequestHTTP(req, clog)
+		resp, respBody, err = h.RequestHTTP(req, clog)
 	} else {
-		resp, _, err = h.RequestHTTPInsecure(req, clog)
+		resp, respBody, err = h.RequestHTTPInsecure(req, clog)
+	}
+
+	if strings.Contains(string(respBody), "Queued") {
+		rc := h.Backend().RedisPool().Get()
+		defer rc.Close()
+		rateLimitKey := fmt.Sprintf("rate_limit:%s", msg.Channel().UUID())
+		rc.Do("SET", rateLimitKey, "engaged")
+
+		// We pause sending 30 seconds so the connection to the SMSC is reset
+		rc.Do("EXPIRE", rateLimitKey, 30)
```

Comment:
SET + EXPIRE = SETNX.. no ?

Reply:
That is deprecated https://redis.io/commands/setnx/

But I can do on line as `rc.Do("SET", rateLimitKey, "engaged", "EX", 30)`